extensions/wikia/ArticleComments/classes/ArticleCommentList.class.php

```diff
@@ -916,63 +924,98 @@ static function formatList($comments) {
 		return $template->render( 'comment-list' );
 	}
 
-	/*
-	 * @access public
-	 * @deprecated don't use it for Oasis, still needed for non-Oasis skins
-	 * @deprecated - not used in Oasis
+	/**
+	 * @deprecated Don't use it for Oasis, still needed for non-Oasis skins
 	 *
 	 * @return String HTML text with rendered comments section
 	 */
-
 	public function render() {
-		return F::app()->renderView('ArticleComments', 'Index');
+		return F::app()->renderView( 'ArticleComments', 'Index' );
  	}
 
 	/**
-	 * static entry point for hook
+	 * Static entry point for hook
 	 *
 	 * @param Title $title
 	 * @param Article $article
 	 *
-	 * @static
-	 * @access public
+	 * @return bool
 	 */
-	static public function ArticleFromTitle( &$title, &$article ) {
-		if( MWNamespace::isTalk($title->getNamespace()) && ArticleComment::isTitleComment($title) ) {
-			global $wgRequest, $wgOut;
-			$redirect = $wgRequest->getText('redirect', false);
-			$diff = $wgRequest->getText('diff', '');
-			$oldid = $wgRequest->getText('oldid', '');
-			$action = $wgRequest->getText('action', '');
-			$permalink = $wgRequest->getInt( 'permalink', 0 );
-
-			if (($redirect != 'no') && empty($diff) && empty($oldid) && ($action != 'history') && ($action != 'delete')) {
-				$parts = ArticleComment::explode($title->getText());
-				$redirectTitle = Title::newFromText($parts['title'], MWNamespace::getSubject($title->getNamespace()));
-				$commentId = $title->getArticleID();
-
-				if ($redirectTitle) {
-					$query = array();
-
-					if ( $permalink || $commentId !== 0 ) {
-						if( $commentId !== 0 ) {
-							/** bugId:11179 @author: nAndy */
-							$permalink = $commentId;
-						}
+	static public function ArticleFromTitle( Title &$title, &$article ) {
+		// Don't bother checking for redirects if we're not loading this article for the current request
+		if ( !self::isTitleForCurrentRequest( $title ) ) {
+			return true;
+		}
 
-						$redirectTitle->setFragment("#comm-$permalink");
-						$page = self::getPageForComment( $redirectTitle, $permalink );
-						if ( $page > 1 ) {
-							$query = array( 'page' => $page );
-						}
-					}
+		// Only handle comments
+		if ( !MWNamespace::isTalk( $title->getNamespace() ) || !ArticleComment::isTitleComment( $title ) ) {
+			return true;
+		}
 
-					$wgOut->redirect($redirectTitle->getFullUrl( $query ));
-				}
+		// See if this is a special request that can't be redirected
+		if ( !self::canSetRedirect() ) {
+			return true;
+		}
+
+		$parts = ArticleComment::explode( $title->getText() );
+		$redirectTitle = Title::newFromText( $parts['title'], MWNamespace::getSubject( $title->getNamespace() ) );
+		if ( empty( $redirectTitle ) ) {
+			return true;
+		}
+
+		$query = [];
+		$commentId = $title->getArticleID();
+		$permalink = F::app()->wg->Request->getInt( 'permalink', 0 );
+
+		// Use comment ID if available - https://wikia.fogbugz.com/f/cases/11179
```

Comment:
We still have a reduced access FogBugz account with all our history, so I'm just making a link out of the possibly confusing number given in the [previous comment](https://github.com/Wikia/app/pull/7940/files#diff-e4fe86c4b1039ccf72823b9e6b3135b9L959).